experimental/builder/include/ck_tile/builder/testing/conv_fwd.hpp

```diff
@@ -0,0 +1,259 @@
+
+    Ops::AElementwiseOp a_elementwise_op;
+    Ops::BElementwiseOp b_elementwise_op;
+    Ops::CDEElementwiseOp cde_elementwise_op;
+
+    // TODO: We shouldn't need to call into an internal namespace here.
+    using Layouts = decltype(factory::internal::GetTensorLayout<SIGNATURE.layout,
+                                                                SPATIAL_DIM,
+                                                                ConvDirection::FORWARD>());
+
+    /// This function returns the `TensorDescriptor` corresponding to
+    /// the input-tensor of the convolution problem. This can then
+    /// be used to, for example, allocate memory.
+    TensorDescriptor<INPUT_TYPE> make_input_descriptor() const
+    {
+        // TODO: We're using old CK functionality to compute the right
+        // values here, mainly because CK tile does not support the
+        // right tensor layouts here. We should probably change that
+        // because CK currently prints an annoying message about it,
+        // plus that would let us get rid of the `to_ck_conv_param()`
+        // function.
+        const auto param = to_ck_conv_param();
+        const auto desc  = ck::utils::conv::make_input_host_tensor_descriptor_g_n_c_wis_packed<
+             typename Layouts::ALayout>(param);
+        return TensorDescriptor<INPUT_TYPE>(desc.GetLengths(), desc.GetStrides());
+    }
+
+    /// This function returns the `TensorDescriptor` corresponding to
+    /// the weight-tensor of  the convolution problem. This can then
+    /// be used to, for example, allocate memory.
+    TensorDescriptor<WEIGHT_TYPE> make_weight_descriptor() const
+    {
+        // See note in implementation of `make_input_descriptor`.
+        const auto param = to_ck_conv_param();
+        const auto desc  = ck::utils::conv::make_weight_host_tensor_descriptor_g_k_c_xs_packed<
+             typename Layouts::BLayout>(param);
+        return TensorDescriptor<WEIGHT_TYPE>(desc.GetLengths(), desc.GetStrides());
+    }
+
+    /// This function returns the `TensorDescriptor` corresponding to
+    /// the output-tensor of the convolution problem. This can then
+    /// be used to, for example, allocate memory.
+    TensorDescriptor<OUTPUT_TYPE> make_output_descriptor() const
+    {
+        // See note in implementation of `make_input_descriptor`.
+        const auto param = to_ck_conv_param();
+        const auto desc  = ck::utils::conv::make_output_host_tensor_descriptor_g_n_k_wos_packed<
+             typename Layouts::ELayout>(param);
+        return TensorDescriptor<OUTPUT_TYPE>(desc.GetLengths(), desc.GetStrides());
+    }
+
+    /// Convert the Args structure into a CK conv_param structure. This
+    /// function is mainly used to be able to use the existing
+    /// CK-functionality to obtain tensor descriptors.
+    ck::utils::conv::ConvParam to_ck_conv_param() const
+    {
+        const auto to_vector = [](const auto& extent) {
+            std::vector<ck::index_t> result;
```

Comment:
This is much simpler:

switch (SPATIAL_DIM){
  case 1: return std::vector<ck::index_t> {extent.width};
  case 2:  return std::vector<ck::index_t> { extent.height, extent.width};
  default: return std::vector<ck::index_t> {extent.depth, extent.height, extent.width};
}

It's silly to optimize away the if-branch but use reserve() and push_back(). If you really want to keep the constexpr-if you can drop the switch-case, but I think the readability is worth a tiny cost in a lambda.